### ISSUE TITLE
Unable to reset password following a forgotten password request

### ISSUE THREAD

**Issue by gazzastone** [LearningLocker/learninglocker#581]
**Version**
1.3.9

**Steps To reproduce**
Click on forgotten password email link and complete Reset Password dialog.

**Expected Result**
Expect password to be changed.

**Actual result**
ErrorException thrown with message "strtotime() expects parameter 1 to be string, array given".

Stacktrace:
```
#25 ErrorException in C:\xampp\htdocs\learninglocker\vendor\laravel\framework\src\Illuminate\Auth\Reminders\DatabaseReminderRepository.php:109
```

**Comment by ryansmith94** [comment]
@gazzastone this is a good one. I'm literally clueless. It looks like you managed to create a user without a `created_at` property which is out of our control.

**Comment by ryansmith94** [comment]
Note to self:
https://github.com/jenssegers/Laravel-MongoDB#auth
https://github.com/jenssegers/Laravel-MongoDB/issues/37

**Comment by ryansmith94** [comment]
I can't reproduce the issue, but when I try to fix it everything breaks.

**Comment by gazzastone** [comment]
ok I will delete this user then recreate it and then do a password reset to see if the issue has gone

**Comment by ryansmith94** [comment]
Ok sounds good.

**Comment by gazzastone** [comment]
so deleted user. registered user ok. requested password reset get the same error

**Comment by ryansmith94** [comment]
No way! That's crazy, I'll see what I can find out. We're all pretty busy over here today, so it may have to wait a little while.

**Comment by gazzastone** [comment]
ok no probs

**Comment by sanduhrs** [comment]
Same issue here, `created_at` actually _is_ an array at least in my setup at 00e3a0137e
```
created_at = {array}[3]
  date = "2015-04-22 13:11:06.000000"
  timezone_type = 3
  timezone = "UTC"
```

**Comment by sanduhrs** [comment]
Using the instructions from #163 fixes the issue for me.
Should this be documented at http://docs.learninglocker.net/installation/ or is it a bug?

**Comment by gazzastone** [comment]
I can confirm that #163 has fixed the issue for me many thanks @sanduhrs

**Comment by ryansmith94** [comment]
Ohhhhhh I'm a complete idiot! My apologies guys! I was well aware of this, but incorrectly assumed you were using the local configs.

@sanduhrs I think what needs to be documented is that you should always use the local config or duplicate the local config and use your own. Using your own just requires you to change the "bootstrap/start.php" file a little.

**Comment by ryansmith94** [comment]
This explains why I couldn't reproduce this issue. We can't add that service provider in a commit because it screws up our continuous integration. However, we could potentially change "bootstrap/start.php" so that it contains...
```php
$env = Helpers::getEnvironment([
  'local' => [gethostname()]
], gethostname());
```

This would ensure that you always used the local config unless you changed this file, which seems ideal to me.

**Comment by gazzastone** [comment]
sounds like a plan to me

**Comment by ryansmith94** [comment]
This issue has now been fixed and the fix has been merged into the [develop branch](https://github.com/LearningLocker/learninglocker/tree/develop). This should be released with version 1.3.10. Thanks @gazzastone and @sanduhrs for all of your help here.
